internal/venice-client-common/src/main/java/com/linkedin/venice/exceptions/VeniceResourceNotFoundException.java

```diff
@@ -0,0 +1,14 @@
+package com.linkedin.venice.exceptions;
+
+/**
+ * Class for all Venice exceptions that are triggered by resources not being found.
+ */
+public class VeniceResourceNotFoundException extends VeniceException {
```

Comment:
Can we be a bit more precise about what the intended usage of this new exception type is? For example, how does it compare and contrast with:

```
VeniceNoStoreException
StoreVersionNotFoundException
```

Would one of the above exceptions match the use case intended for this new one?

Or should the above exceptions extend from this new one?

Reply:
Yes, I've considered using them but the naming isn't a good fit since this is a partition-level.
Actually it can be a good idea to extend `VeniceNoStoreException` to `StoreVersionNotFoundException` and `StorePartitionNotFoundException`(the new one going to throw here). Will make some changes